src/Bindings/ManualBindings_World.cpp

```diff
@@ -380,9 +378,9 @@ static int tolua_cWorld_QueueTask(lua_State * tolua_S)
 		return cManualBindings::lua_do_error(tolua_S, "Error in function call '#funcname#': Could not get function reference of parameter #1");
 	}
 
-	auto task = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
-	Plugin->AddResettable(task);
-	self->QueueTask(task);
+	auto ResettableTask = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
+	Plugin->AddResettable(ResettableTask);
+	self->QueueTask(std::bind(&cLuaWorldTask::Run, ResettableTask, std::placeholders::_1));
```

Comment:
This is the problem with functional code in imperative languages.
Maybe as SphinxC0re suggested on IRC what we really want is to be able to go full FRP with something like:
```
auto Task = self->QueueTask();
Task.flat_map([&Resettable](cWorld & a_World)
    {
         return Resettable.TryDoWithPlugin();
    })
    .zip(Task)
    .then([=FnRef](std::tuple<cPlugin, cWorld & a_args)
    {
         (a_args.first)->Call(FnRef, a_args.second);
    });